core/rust/src/lib.rs

```diff
@@ -8,14 +8,18 @@ use std::{
 pub extern "C" fn Cairo0ClassHash(class_json_str: *const c_char, hash: *mut c_uchar) {
     let class_json = unsafe { CStr::from_ptr(class_json_str) }.to_str().unwrap();
     let class: Result<LegacyContractClass, serde_json::Error> = serde_json::from_str(class_json);
-    if class.is_err() {
-        return;
-    }
-    let class_hash = class.unwrap().class_hash();
-    if class_hash.is_err() {
-        return;
-    }
+
+    let class = match class {
+        Ok(c) => c,
+        Err(_) => return,
+    };
+
+    let class_hash = match class.class_hash() {
+        Ok(h) => h,
+        Err(_) => return,
+    };
+
     let hash_slice: &mut [u8] = unsafe { slice::from_raw_parts_mut(hash, 32) };
-    let hash_bytes = class_hash.unwrap().to_bytes_be();
```

Comment:
but why does it work without `unwrap()` now?

Reply:
These might be helpful

https://doc.rust-lang.org/std/result/enum.Result.html

https://doc.rust-lang.org/std/result/enum.Result.html#method.and_then

https://doc.rust-lang.org/std/result/enum.Result.html#method.or_else

Reply:
hash_bytes was a Result, became a FieldElement
When it was a Result, we could use unwrap to get FieldElement.
Now we don't need to use it because it is now a FieldElement.

Reply:
Ah sorry I didn't notice match above